### PR TITLE
Model details, ORM models, schemas, endpoints

### PR DESCRIPTION
**Pull request by HermanPlay**
For now, endpoints are implemented the same way as we implemented every other one.

### CODE REVIEW

File: app/api/models/model_details.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+)
+from sqlalchemy.orm import relationship
+
+from models.base_class import Base
+
+
+class ModelDetails(Base):
+    id = Column(Integer, primary_key=True)
+    model_id = Column(Integer, ForeignKey("model.id"))
+    artifact_uri = Column(String(255), nullable=False)
```
Comment (matined) [inline app/api/models/model_details.py:15]: all the fields beside ids are rather nullable, we can register a model withput specifing cpu, memory itp

File: app/api/routers/model.py [hunk 2]
```diff
@@ -7,9 +7,11 @@
 
 from schemas import model as model_schemas
 from services import ModelService, get_db
+from routers.model_details import router as model_details_router
 
 
 router = APIRouter(prefix="/model", tags=["model"])
+router.include_router(model_details_router, prefix="/{model_id}/details")
```
Comment (matined) [inline app/api/routers/model.py:14]: is there a reason to put prefix here instead of in the construction?

File: app/api/routers/model_details.py [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+    """
+    model_details = ModelDetailsService.get_model_details_by_id(
+        db=db, model_details_id=model_details_id
+    )
+    if not model_details:
+        raise HTTPException(status_code=404, detail="ModelDetails not found!")
+    return model_details
+
+
+@router.get("/", response_model=list[ModelDetails], status_code=200, deprecated=True)
```
Comment (matined) [inline app/api/routers/model_details.py:31]: we dont want this endpoint, we dont need to list all model details

File: app/api/routers/model_details.py [hunk 4]
```diff
@@ -0,0 +1,145 @@
+from fastapi import APIRouter, Query, Depends, HTTPException
+from sqlalchemy.orm import Session
+
+from schemas.model_details import ModelDetails, PutModelDetails, PatchModelDetails
+from services import ModelDetailsService, get_db
+
+router = APIRouter(tags=["model_details"])
+
+
+@router.get("/{model_details_id}", response_model=ModelDetails, status_code=200)
```
Comment (matined) [inline app/api/routers/model_details.py:10]: I'd prefer to get details only by model_id

File: app/api/schemas/model_details.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    replicas: Annotated[int, Field(description="Number of replicas")]
+    cpu_request: Annotated[str, Field(description="CPU request")]
+    cpu_limit: Annotated[str, Field(description="CPU limit")]
+    memory_request: Annotated[str, Field(description="Memory request")]
+    memory_limit: Annotated[str, Field(description="Memory limit")]
+
+
+class PutModelDetails(ModelDetailsBase):
+    model_id: Annotated[int, Field(description="Model ID")]
+    pass
```
Comment (matined) [inline app/api/schemas/model_details.py:17]: it a debris, right?

**Review by matined — APPROVED** [review]

File: app/api/routers/model_details.py [hunk 6]
```diff
@@ -0,0 +1,145 @@
+from fastapi import APIRouter, Query, Depends, HTTPException
+from sqlalchemy.orm import Session
+
+from schemas.model_details import ModelDetails, PutModelDetails, PatchModelDetails
+from services import ModelDetailsService, get_db
+
+router = APIRouter(tags=["model_details"])
```
Comment (matined) [inline app/api/routers/model_details.py:7]: let's use '-' instead of '_' in tags

**Review by matined — CHANGES_REQUESTED** [review]

File: app/api/schemas/model_details.py (hunk 5 above)
Comment (HermanPlay) [reply app/api/schemas/model_details.py:17]: I was thinking that we cannot modify model_id, and we can only specify it when we add model_details. If I am thinking wrong we can make a possibility to modify model_id

Comment (matined) [reply app/api/schemas/model_details.py:17]: My comment was about "pass", but let me describe the general idea which is that every time a new model is created a new record is inserted into model_details and it's a one-to-one relationship. By default all values beside model_id are nulls, but you have to insert them before deploying a model.

**Review by matined — CHANGES_REQUESTED** [review]
I'm sorry, but I still need some changes:

1. I don't want to pass model_details_id while calling endpoints, only model_id
2. I don't want to expose PUT and DELETE 

So, finally, We want to expose two endpoints:
- GET /model/{model_id}/details
- PATCH /model/{model_id}/details

put and delete should be performed when PUT /model and DELETE /model/{model_id} are called

**Review by matined — APPROVED** [review]

**Comment by matined** [comment]
> Maybe now it works :)

Great job!

**Review by matined — APPROVED** [review]

**Pull request merged**